pkg/ingester/flush.go

```diff
@@ -200,11 +204,18 @@ func (i *Ingester) shouldFlushChunk(c *desc, fp model.Fingerprint) flushReason {
 		return reasonAged
 	}
 
-	// Chunk should be flushed if their last update is older then MaxChunkIdle
+	// Chunk should be flushed if their last update is older then MaxChunkIdle.
 	if model.Now().Sub(c.LastUpdate) > i.cfg.MaxChunkIdle {
 		return reasonIdle
 	}
 
+	// A chunk that has a stale can be flushed if possible.
+	if i.cfg.MaxStaleChunkIdle > 0 &&
+		value.IsStaleNaN(float64(c.LastValue)) &&
+		model.Now().Sub(c.LastUpdate) > i.cfg.MaxStaleChunkIdle {
```

Comment:
Looking at this, we would reach here only if `MaxChunkIdle > MaxStaleChunkIdle`. So in that case if staleness is observed before `MaxChunkIdle` is reached, why not just check for staleness instead of additional `MaxStaleChunkIdle`?

Reply:
We don't want to flush immediately for just one scrape failure. And it doesn't make sense to set `MaxChunkIdle < MaxStaleChunkIdle`

Reply:
Isn't a series marked stale after like 5 minutes of no samples and not after a single scrape failure?

Reply:
Brian has you covered https://www.robustperception.io/staleness-and-promql

Reply:
Oh, my bad, so the 5m comes from something like `MaxStaleChunkIdle` inside Prometheus. I thought the stale sample was added after 5m.